source/frontend/src/features/leases/models/LeaseStatusUpdateSolver.ts

```diff
@@ -0,0 +1,17 @@
+import { IUpdateCompensationStrategy } from '@/features/mapSideBar/compensation/models/UpdateCompensationStrategy';
+import { ApiGen_Base_CodeType } from '@/models/api/generated/ApiGen_Base_CodeType';
+
+export class LeaseStatusUpdateSolver implements IUpdateCompensationStrategy {
+  constructor(private readonly fileStatus: ApiGen_Base_CodeType<string> | null = null) {
+    this.fileStatus = fileStatus;
+  }
+
+  // eslint-disable-next-line unused-imports/no-unused-vars
+  canEditOrDeleteCompensation(isDraftCompensation: boolean): boolean {
+    if (this.fileStatus === null) {
```

Comment:
Do we have rules for this? this seems more permissive then I'd expect.

Reply:
Not at this moment, already brought to BAs attention.